src/farcaster/types.ts

```diff
@@ -0,0 +1,19 @@
+/**
+ * Frame address type to determine if the client is requesingt  verified addresses or a custody address
+ *  Note: exported as public Type
+ */
+
+export enum FarcasterAddressType {
+  VerifiedAddresses,
+  CustodyAddress,
+}
+
+/**
+ * GetFarcasterUserAddressesRequest
+ *
+ * Note: exported as public Type
+ */
+export interface GetFarcasterUserAddressesRequest {
```

Comment:
Where can we see an example of this kind of request?

Reply:
This can be used to display the user's address.
In case of searchcaster they wrote the code [here](https://github.com/gskril/searchcaster/blob/03fb485d8dbe52a2248d3353dfc2f15164caa949/src/pages/profiles.js#L135) and [here](https://github.com/gskril/searchcaster/blob/03fb485d8dbe52a2248d3353dfc2f15164caa949/src/pages/api/profiles.js#L141) to show the user's profile

Reply:
Ok so is this the body we are getting?
```
body: {
        id: p.id,
        address: p?.owner || null,
        username: p.username,
        displayName: p.display_name,
        bio: p.bio,
        followers: p.followers,
        following: p.following,
        avatarUrl: p.avatar_url,
        isVerifiedAvatar: p.avatar_verified,
        registeredAt: new Date(p.registered_at).getTime(),
      },
      connectedAddress:
        p.verifications && p.verifications[0]
          ? p.verifications[0].address
          : connected_address || null,
      connectedAddresses:
        p.verifications && p.verifications[0]
          ? p.verifications.map((v) => v.address)
          : [],
```

and from where this request is coming?

I guess I am still not sure the end-to-end flow. Like:
Step 1. the body comes from this scenario
Step 2. we use the body in this way
Step 3. we have the address

Reply:
We created this type to ask the user whether they want a given user's `custody` or `verified address`
For verified address, we are using the api [here](https://docs.neynar.com/reference/verifications-v1)
For custody address, we are using the api [here](https://docs.neynar.com/reference/custody-address-v1)

What I was pointing to in `searchcaster` is the use case of `getFasrcasterUserAddress`. 

So if I am a `searchcaster` dev and I want to use `onchainkit` to get a user's address to build a profile object. I can do this (sudo code)

```
const userProfile =  {
 custodyAddress: getFarcasterUserAddress({fid:12, FarcasterAddressType.CustodyAddress}, {neynarAPIKey: '123'})
 verifiedAddresses: getFarcasterUserAddress({fid:12, FarcasterAddressType.VerifiedAddresses})
}
```